Automattic/wp-calypso
# Checkout: Editor checkout modal.

#### Changes proposed in this Pull Request

* Adds editor checkout modal.

This PR supercedes https://github.com/Automattic/wp-calypso/pull/45902 with the following differences:
- Calling the checkout modal is done through `window.wp.hooks.doAction`.
- `AsyncLoad` for checkout modal.
- Passing of the entire cart data as an object.
- The checkout modal is placed inside `client/blocks/editor-checkout-modal`.
- Hiding behind feature flag `gutenboarding/checkout-overlay`.

#### Testing instructions

* Sandbox widgets.wp.com.
* Run `yarn dev --sync` on `apps/wpcom-block-editor`.
* Open block editor of the free site with the feature flag `?flags=post-editor/checkout-overlay`, e.g. `calypso.localhost:3000/block-editor/post/your_sandboxed_free_site.wordpress.com?flags=post-editor/checkout-overlay`
* Open up dev console, change the frame to `post.php`, and run the following code.
```js
let cartData = { products: [{
    product_id: 1009,
    product_slug: 'personal-bundle',
}] };

window.wp.hooks.doAction('a8c.wpcom-block-editor.openCheckoutModal', cartData);
```
* Checkout modal should appear.



## Screenshot

![2020-10-07_11-46-14 (1)](https://user-images.githubusercontent.com/1287077/95308778-fabefc80-0892-11eb-80d9-351e2962ae45.gif)
